conan_provider.cmake

```diff
@@ -50,9 +50,13 @@ function(detect_os OS OS_API_LEVEL OS_SDK OS_SUBSYSTEM OS_VERSION)
                 message(STATUS "CMake-Conan: cmake_osx_sysroot=${CMAKE_OSX_SYSROOT}")
                 set(${OS_SDK} ${_OS_SDK} PARENT_SCOPE)
             endif()
-            if(DEFINED CMAKE_OSX_DEPLOYMENT_TARGET)
+            if(DEFINED CONAN_CUSTOM_CMAKE_OSX_DEPLOYMENT_TARGET)
                 message(STATUS "CMake-Conan: cmake_osx_deployment_target=${CMAKE_OSX_DEPLOYMENT_TARGET}")
                 set(${OS_VERSION} ${CMAKE_OSX_DEPLOYMENT_TARGET} PARENT_SCOPE)
+                # Remove the CMAKE_OSX_DEPLOYMENT_TARGET from the CACHE to avoid errors
+                if($CACHE{CMAKE_OSX_DEPLOYMENT_TARGET})
+                    unset(CMAKE_OSX_DEPLOYMENT_TARGET CACHE)
+                endif()
```

Comment:
The idea is to keep the CACHE clean so that we take the value of the variable only if it is passed as a parameter by the user. I don't know if this is exactly the expected use case.

Reply:
This reads a bit concerning in my opinion, I don't think that the conan provider should be altering in any way the user defined CMAKE_xxxx variables.